apps/opik-backend/src/main/java/com/comet/opik/domain/ProjectMetricsDAO.java

```diff
@@ -0,0 +1,79 @@
+package com.comet.opik.domain;
+
+import com.comet.opik.api.DataPoint;
+import com.comet.opik.api.metrics.ProjectMetricRequest;
+import com.comet.opik.infrastructure.instrumentation.InstrumentAsyncUtils;
+import com.google.inject.ImplementedBy;
+import io.r2dbc.spi.Connection;
+import io.r2dbc.spi.Result;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.reactivestreams.Publisher;
+import org.stringtemplate.v4.ST;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+import static com.comet.opik.domain.AsyncContextUtils.bindWorkspaceIdToMono;
+import static com.comet.opik.infrastructure.instrumentation.InstrumentAsyncUtils.endSegment;
+import static com.comet.opik.infrastructure.instrumentation.InstrumentAsyncUtils.startSegment;
+import static com.comet.opik.utils.AsyncUtils.makeMonoContextAware;
+
+@ImplementedBy(ProjectMetricsDAOImpl.class)
+public interface ProjectMetricsDAO {
+    Mono<List<DataPoint>> getTraceCount(UUID projectId, ProjectMetricRequest request, Connection connection);
+}
+
+@Slf4j
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+class ProjectMetricsDAOImpl implements ProjectMetricsDAO {
+    private static final String GET_TRACE_COUNT = """
+            SELECT toStartOfInterval(start_time, toIntervalHour(1)) AS bucket,
+                   count() as count
+            FROM traces
+            WHERE project_id = :project_id
+                AND workspace_id = :workspace_id
+                AND start_time > parseDateTime64BestEffort(:start_time, 9)
+                AND end_time \\< parseDateTime64BestEffort(:end_time, 9)
```

Comment:
Shouldn't this be like:

```suggestion
                AND start_time >= parseDateTime64BestEffort(:start_time, 9)
                AND end_time <= parseDateTime64BestEffort(:end_time, 9)
```

Reply:
Done. The escaping is crucial because without it `StringTemplate` breaks